### PR TITLE
added public facing analysis for sql instance users and datasets

### PR DESCRIPTION
**Pull request by Aryan-code-dev**

### CODE REVIEW

**Review by ganeshjagdale9545 — APPROVED** [review]
Looks good

**Review by ganeshjagdale9545 — CHANGES_REQUESTED** [review]
Implement cleanup query for `GCPSQLUser` and `GCPSQLDatabase`

**Comment by ganeshjagdale9545** [comment]
> Implement cleanup query for `GCPSQLUser` and `GCPSQLDatabase`

@Aryan-code-dev take look on this comment

**Comment by Aryan-code-dev** [comment]
> Implement cleanup query for `GCPSQLUser` and `GCPSQLDatabase`

okay

**Comment by Aryan-code-dev** [comment]
> Implement cleanup query for `GCPSQLUser` and `GCPSQLDatabase`

clean up query already exists in gcp_sql_cleanup.json

File: cartography/data/jobs/analysis/gcp_sql_instance_analysis.json [hunk 1]
```diff
@@ -5,9 +5,27 @@
             "iterative": true,
             "iterationsize": 1000
         },
+        {
+            "query": "MATCH (u:GCPSQLUser{exposed_internet:true})<-[r:USED_BY]-(:GCPSQLInstance})<-[:RESOURCE]-(:GCPProject{id: $GCP_PROJECT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WITH u LIMIT $LIMIT_SIZE REMOVE u.exposed_internet, u.exposed_internet_type return COUNT(*) as TotalCompleted",
```
Comment (mpurusottamc) [inline cartography/data/jobs/analysis/gcp_sql_instance_analysis.json:9]: public facing logic does not apply to GCPSQLUser object. Please remove this code from this file.

File: cartography/data/jobs/analysis/gcp_sql_instance_analysis.json [hunk 2]
```diff
@@ -5,9 +5,27 @@
...
+            "query": "MATCH (d:GCPSQLDatabase{exposed_internet:true})<-[r:HAS_DATABASE]-(:GCPSQLInstance)<-[:RESOURCE]-(:GCPProject{id: $GCP_PROJECT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) WITH d LIMIT $LIMIT_SIZE REMOVE d.exposed_internet, d.exposed_internet_type return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 1000
+        },
         {
             "query": "MATCH (i:GCPSQLInstance)<-[:RESOURCE]-(:GCPProject{id: $GCP_PROJECT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) \nWHERE '0.0.0.0/0' IN i.authorized_networks \nSET i.exposed_internet = true, i.exposed_internet_type = coalesce(i.exposed_internet_type, []) + 'unrestricted_access'",
             "iterative": false
+        },
+        {
+            "query": "MATCH (u:GCPSQLUser)<-[r:USED_BY]-(i:GCPSQLInstance)<-[:RESOURCE]-(:GCPProject{id: $GCP_PROJECT_ID})<-[:OWNER]-(:CloudanixWorkspace{id: $WORKSPACE_ID}) \nWHERE i.exposed_internet = true \nSET u.exposed_internet = true, u.exposed_internet_type = i.exposed_internet_type",
```
Comment (mpurusottamc) [inline cartography/data/jobs/analysis/gcp_sql_instance_analysis.json:23]: same here as well.

**Review by mpurusottamc — CHANGES_REQUESTED** [review]
one minor change is needed.

**Review by mpurusottamc — APPROVED** [review]
merging it. will do the changes before deployment.

**Pull request merged**